# Subject: Backport Go 1.26 printf vet compatibility
#
# Upstream source:
# https://gitlab.com/gitlab-org/gitlab-runner/-/commit/092b67b35513881689a0b6faf87dd738e6f856af
#
# Backport the Warningf corrections needed to compile GitLab Runner 18.11.4
# with the Go 1.26 module directive required by golang.org/x/crypto v0.56.0.
diff --git a/shells/abstract.go b/shells/abstract.go
index f755c18e0..ec96ba30a 100644
--- a/shells/abstract.go
+++ b/shells/abstract.go
@@ -235,7 +235,7 @@ func (b *AbstractShell) cacheExtractor(
 
 		cacheConfig, warning, err := newCacheConfig(info.Build, cacheOptions.Key)
 		if warning != "" {
-			w.Warningf(warning)
+			w.Warningf("%s", warning)
 		}
 		if err != nil {
 			w.Noticef("Skipping cache extraction due to %v", err)
@@ -291,7 +291,7 @@ func (b *AbstractShell) extractCacheOrFallbackCachesWrapper(
 		}
 		cc, warning, err := newCacheConfig(build, key, keyChecks...)
 		if warning != "" {
-			w.Warningf(warning)
+			w.Warningf("%s", warning)
 		}
 		if err != nil {
 			w.Noticef("Skipping cache extraction due to %v", err)
@@ -561,7 +561,7 @@ func (b *AbstractShell) writeCacheExports(w ShellWriter, variables map[string]st
 func (b *AbstractShell) writeGitSSLConfig(w ShellWriter, build *common.Build, where []string) {
 	host, err := b.getRemoteHost(build)
 	if err != nil {
-		w.Warningf("git SSL config: Can't get repository host. %w", err)
+		w.Warningf("git SSL config: Can't get repository host. %v", err)
 		return
 	}
 
@@ -1355,7 +1355,7 @@ func (b *AbstractShell) archiveCache(
 
 		cacheConfig, warning, err := newCacheConfig(info.Build, cacheOptions.Key)
 		if warning != "" {
-			w.Warningf(warning)
+			w.Warningf("%s", warning)
 		}
 		// Skip archiving if no cache is defined
 		if err != nil {
